core/harambe_core/parser/type_price.py

```diff
@@ -0,0 +1,82 @@
+import re
+from typing import Any, Optional, Union
+
+from price_parser import Price
+from pydantic import BeforeValidator
+from typing_extensions import Annotated, Literal
+
+from harambe_core.parser.constants import CURRENCY_MAP, PRICE_NOT_AVAILABLE_PHRASES
+from harambe_core.parser.type_currency import ParserTypeCurrency
+
+# this will capture only the amount
+number_pattern = re.compile(r"-?.?\d[\d,-\.]*")
+
+
+class ParserTypePrice:
+    def __new__(cls, required: bool = True) -> Any:
+        validator = BeforeValidator(cls.validate_price)
+        base = dict[
+            Literal["currency", "raw_currency", "amount", "raw_price"],
+            Optional[Union[str, float]],
+        ]
+
+        return Annotated[
+            base if required else base | None,
+            validator,
+        ]
+
+    @staticmethod
+    def validate_price(
+        value: Union[str, float, int, None],
+    ) -> Optional[
+        dict[
+            Literal["currency", "raw_currency", "amount", "raw_price"],
+            Optional[Union[str, float]],
+        ]
+    ]:
```

Comment:
since this is a before validator we shouldnt throw an error here. 

If we can't convert the string into a dict. We should be returning the original string, letting pydantic throw the error. Thus keeping all our validation errors similar in error structure.

See how its done in the enum parser

Reply:
Do we need to throw an error in this case?
We probably can still parse the value out of it because we use regex?